Sources/ManageSettings.php

```diff
@@ -2275,4 +2281,293 @@ function ModifyAlertsSettings()
+		WHERE a.variable like {string:avar}
+		GROUP BY a.variable
+		ORDER BY a.variable desc',
+		array(
+			'policy_approved' => 'policy_approved',
+			'avar' => 'policy_text%',
+		)
+	);
+	
+	$context['poc']['policy_management'] = array();
+	while ($row = $smcFunc['db_fetch_assoc']($request))
+	{
+		$context['poc']['policy_management'][] = array(
+			'name' => $row['variable'],
+			'amount' => $row['amount'],
+			'new' => ($row['variable'] == 'policy_text' . $currentVersion ? true : false),
+		);
+	}
+	$smcFunc['db_free_result']($request);
+	if ($return_config)
+		return $config_vars;
+
+	// Saving?
+	if (isset($_GET['save']))
+	{
+		checkSession();
+
+		call_integration_hook('integrate_save_policy_settings');
+
+		if (!empty($_REQUEST['enforce_new']))
+		{
+			if ($smcFunc['db_title'] == 'MySQLi')
+				$select = '
+					UPDATE {db_prefix}themes c
+					JOIN (
+						SELECT a.id_member
+						FROM {db_prefix}themes a
+						LEFT JOIN {db_prefix}themes b ON (a.id_member = b.id_member and b.variable = {string:bvar})
+						WHERE a.variable = {string:avar} and a.value = {string:aval} 
+							AND ( b.value != {string:bval}  or b.value is null)
+					) d ON (d.id_member = c.id_member)
+					SET c.value = {string:value}
+					WHERE c.variable = {string:avar}';
+			else
+				$select = '
+					UPDATE {db_prefix}themes
+					SET value = {string:value}
+					WHERE EXISTS (
+						SELECT a.id_member
+						FROM {db_prefix}themes a
+						LEFT JOIN {db_prefix}themes b ON (a.id_member = b.id_member and b.variable = {string:bvar})
+						WHERE a.variable = {string:avar} and a.value = {string:aval} 
+							AND ( b.value != {string:bval}  or b.value is null)
+							AND a.id_member = {db_prefix}themes.id_member
+					)
+					AND variable = {string:avar}';
+			
+			$smcFunc['db_query']('',
+				$select,
+				array(
+					'value' => '0',
+					'bvar' => 'policy_approved',
+					'avar' => 'policy_isvalid',
+					'aval' => '1',
+					'bval' => 'policy_text' . $currentVersion,
+				)
+			);
+		}
+		elseif (!empty($_REQUEST['save_setting']))
+		{
+			unset($config_vars[1]);
+			saveDBSettings($config_vars);
+		}
+		elseif (!empty($_REQUEST['save_new_policy']) || !empty($_REQUEST['update_policy']))
+		{
+			if (!empty($_REQUEST['save_new_policy']))
+				$currentVersion++;
+			
+			$config_vars[] = array('text', 'policy_text'.$currentVersion);
+			$_POST['policy_text'.$currentVersion] = $_REQUEST['policy_text'];
+			$config_vars[] = array('text', 'policy_version');
+			$_POST['policy_version'] = 'policy_text'.$currentVersion;
+			unset($config_vars[1]);
+			saveDBSettings($config_vars);
+		}
+
+		$_SESSION['adm-save'] = true;
+		redirectexit('action=admin;area=featuresettings;sa=policy');
+	}
+	elseif (isset($_GET['manage']))
+	{
+		checkSession();
+		
+		call_integration_hook('integrate_manage_policy_settings');
+		
+		// set user with the policy unvalid
+			if ($smcFunc['db_title'] == 'MySQLi')
```

Comment:
I'd rather avoid having two versions of the queries in this section. If necessary, feel free to create another `$smcFunc['db_***']` function to handle this cleanly.

Reply:
Make less sense to place sutch special query in a generic env,
this query is only here needed.

Reply:
got fixed.